src/transformers/models/parakeet/modeling_parakeet.py

```diff
@@ -0,0 +1,602 @@
+        self.activation = ACT2FN[config.hidden_act]
+        self.linear2 = nn.Linear(config.intermediate_size, config.hidden_size, bias=config.use_bias)
+        self.activation_dropout = config.activation_dropout
+
+    def forward(self, hidden_states):
+        hidden_states = self.activation(self.linear1(hidden_states))
+        hidden_states = nn.functional.dropout(hidden_states, p=self.activation_dropout, training=self.training)
+        hidden_states = self.linear2(hidden_states)
+        return hidden_states
+
+
+class ParakeetEncoderConvModule(nn.Module):
+    def __init__(self, config: ParakeetEncoderConfig):
+        super().__init__()
+        self.padding = (config.conv_kernel_size - 1) // 2
+
+        self.pointwise_conv1 = nn.Conv1d(
+            config.hidden_size, config.hidden_size * 2, kernel_size=1, bias=config.use_bias
+        )
+        self.depthwise_conv = nn.Conv1d(
+            config.hidden_size,
+            config.hidden_size,
+            kernel_size=config.conv_kernel_size,
+            padding=0,
+            groups=config.hidden_size,
+            bias=config.use_bias,
+        )
+        self.batch_norm = nn.BatchNorm1d(config.hidden_size)
+        self.activation = ACT2FN[config.hidden_act]
+        self.pointwise_conv2 = nn.Conv1d(config.hidden_size, config.hidden_size, kernel_size=1, bias=config.use_bias)
+
+    def forward(self, hidden_states, attention_mask=None):
+        hidden_states = hidden_states.transpose(1, 2)
+        hidden_states = self.pointwise_conv1(hidden_states)
+        hidden_states = nn.functional.glu(hidden_states, dim=1)
+
+        # Apply padding mask before convolution
+        if attention_mask is not None:
+            all_masked_rows = torch.all(~attention_mask, dim=-1)
+            hidden_states = hidden_states.masked_fill(all_masked_rows, 0.0)
+
+        hidden_states = nn.functional.pad(hidden_states, (self.padding, self.padding))
+        hidden_states = self.depthwise_conv(hidden_states)
+        hidden_states = self.batch_norm(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        hidden_states = self.pointwise_conv2(hidden_states)
+
+        return hidden_states.transpose(1, 2)
+
+
+def eager_attention_forward(
+    module: nn.Module,
+    query: torch.Tensor,
+    key: torch.Tensor,
+    value: torch.Tensor,
+    attention_mask: Optional[torch.Tensor],
+    scaling: float,
+    dropout: float = 0.0,
+    **kwargs: Unpack[TransformersKwargs],
+):
+    attn_weights = torch.matmul(query, key.transpose(2, 3)) * scaling
+    if attention_mask is not None:
+        attn_weights = attn_weights + attention_mask
+
+    attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(query.dtype)
+    attn_weights = nn.functional.dropout(attn_weights, p=dropout, training=module.training)
+    attn_output = torch.matmul(attn_weights, value)
+    attn_output = attn_output.transpose(1, 2).contiguous()
+
+    return attn_output, attn_weights
```

Comment:
using modular this can be imported

Reply:
@eustlb had a [modular script](https://github.com/eustlb/transformers/blob/add_fastconformer-changes/src/transformers/models/parakeet/modular_parakeet.py) in his fork? Maybe he had a reason for removing it when pushing changes here?

Anyway I just had to add `self.num_key_value_groups` to `ParakeetEncoderAttention` to be able to use `eager_attention_forward` from `src/transformers/models/llama/modeling_llama.py`

But I agree, there's probably a cleaner way to have `ParakeetEncoderAttention` copied from `LlamaAttention` with modular...